Polygon_mesh_processing/test/Polygon_mesh_processing/delaunay_remeshing_test.cpp

```diff
@@ -0,0 +1,112 @@
+    : CGAL::data_file_path("meshes/anchor_dense.off");
+
+  Mesh mesh;
+  if (!PMP::IO::read_polygon_mesh(filename, mesh) || !CGAL::is_triangle_mesh(mesh))
+  {
+    std::cerr << "Invalid input." << std::endl;
+    return 1;
+  }
+
+  double target_edge_length = (argc > 2) ? std::stod(std::string(argv[2])) : 0.1;
+  double fdist = (argc > 3) ? std::stod(std::string(argv[3])) : 0.01;
+  double sharp_angle = (argc > 4) ? std::stod(std::string(argv[4])) : 45.;
+
+  std::cout << "Detect features..." << std::endl;
+
+  // automatically detect features
+  using EIFMap = boost::property_map<Mesh, CGAL::edge_is_feature_t>::type;
+  EIFMap eif = get(CGAL::edge_is_feature, mesh);
+  PMP::detect_sharp_edges(mesh, sharp_angle, eif);
+
+  // collect features in a vector of segments
+  auto vpmap = get_property_map(boost::vertex_point, mesh);
+  std::vector<std::vector<Point> > segments;
+  for (auto e : edges(mesh))
+  {
+    if (get(eif, e))
+    {
+      auto ps = get(vpmap, source(halfedge(e, mesh), mesh));
+      auto pt = get(vpmap, target(halfedge(e, mesh), mesh));
+      std::vector<Point> s = { ps, pt };
+      if (s[1] > s[0]) std::swap(s[0], s[1]);
+      segments.push_back(s);
+    }
+  }
+  std::cout << "Detect features done." << std::endl;
+
+  std::cout << "Start remeshing of " << filename
+    << " (" << num_faces(mesh) << " faces)..." << std::endl;
+
+  std::cout << "***********************" << std::endl;
+  std::cout << "Remeshing with edge_is_constrained_map..." << std::endl;
+
+  Mesh outmesh1 = PMP::surface_Delaunay_remeshing(mesh,
+    PMP::parameters::protect_constraints(true)
+    .mesh_edge_size(target_edge_length)
+    .mesh_facet_size(target_edge_length)
+    .mesh_facet_distance(fdist)
+    .edge_is_constrained_map(eif));
+
+  std::cout << "Remeshing with edge_is_constrained_map done." << std::endl;
+
+  std::cout << "***********************" << std::endl;
+  std::cout << "Remeshing with polyline_constraints..." << std::endl;
+
+  Mesh outmesh2 = PMP::surface_Delaunay_remeshing(mesh,
+    PMP::parameters::protect_constraints(true)
+    .mesh_edge_size(target_edge_length)
+    .mesh_facet_size(target_edge_length)
+    .mesh_facet_distance(fdist)
+    .polyline_constraints(segments));
+
+  std::cout << "Remeshing with polyline_constraints done." << std::endl;
+
+  // Check
+  const std::size_t nbv1 = std::distance(outmesh1.vertices().begin(), outmesh1.vertices().end());
+  const std::size_t nbv2 = std::distance(outmesh2.vertices().begin(), outmesh2.vertices().end());
+  assert(nbv1 == nbv2);
+
+  const std::size_t nbf1 = std::distance(outmesh1.faces().begin(), outmesh1.faces().end());
+  const std::size_t nbf2 = std::distance(outmesh2.faces().begin(), outmesh2.faces().end());
+  assert(nbf1 == nbf2);
+
+  std::cout << "***********************" << std::endl;
+  std::cout << "Remeshing with features_angle_bound..." << std::endl;
+  Mesh outmesh3 = PMP::surface_Delaunay_remeshing(mesh,
```

Comment:
@lrineau I am wondering why `outmesh3` is different from the two others.
I can't find what could make a difference in the meshing process, for now

Reply:
The nb of vertices after protecting balls placement is the same, but nb of finite facets is different

Reply:
That took me more than two days, but I eventually found the solution: https://github.com/CGAL/cgal/pull/6084/commits/83fb998a6621a129256d0b9fc896c3515c1db87a!